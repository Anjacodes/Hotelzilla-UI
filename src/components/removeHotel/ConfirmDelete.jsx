import React from 'react';
import PropTypes from 'prop-types';

const ConfirmDelete = ({ visible, onOk, onCancel, hotel }) => {
  const handleVisible = () => {
    if (visible) return 'block ';
    else return 'hidden ';
  };

  return (
    <div
      className={
        handleVisible() + 'bg-black/20 absolute top-0 left-0 w-screen h-screen'
      }>
      <section className="bg-slate-50 mt-[10%] mx-[10%] xl:mx-[25%] flex flex-col relative ">
        <h3 className="font-Taxicab text-xl border-b-2 px-4 py-2">
          Please confirm:
        </h3>
        <i
          class="fa-solid fa-xmark absolute top-2 right-4 text-xl hover:text-red-600"
          onClick={onCancel}></i>

        <p className="px-6 py-4 text-black text-lg">
          Are you sure you want to delete {hotel.name}?
        </p>

        <div className="flex gap-4 py-4 justify-end mt-6 px-4 border-t-2">
          <button
            className="px-4 py-2 rounded-md text-white hover:bg-green-300 hover:text-green-800 font-semibold bg-green-600"
            type="button"
            onClick={onOk}>
            Sure
          </button>
          <button
            className="px-4 py-2 rounded-md text-white hover:bg-red-300 hover:text-red-800 font-semibold bg-red-600"
            type="button"
            onClick={onCancel}>
            Cancel
          </button>
        </div>
      </section>
    </div>
  );
};

ConfirmDelete.propTypes = {};

export default ConfirmDelete;
