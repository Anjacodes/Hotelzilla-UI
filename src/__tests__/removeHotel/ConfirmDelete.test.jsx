/**
 * @vitest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { it, describe, expect } from 'vitest';
import store from '../../redux/configStore';
import ConfirmDelete from '../../components/removeHotel/ConfirmDelete';

describe('Confirm Delete', () => {
  const hotel = {
    description: 'Test',
    id: 77,
    image: null,
    name: 'Test',
    rating: 3,
    city: { id: 1, name: 'Argentina' },
  };
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <ConfirmDelete hotel={hotel} />
        </BrowserRouter>
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
