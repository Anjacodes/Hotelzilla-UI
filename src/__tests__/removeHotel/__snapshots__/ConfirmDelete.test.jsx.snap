// Vitest Snapshot v1

exports[`Confirm Delete > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hidden absolute top-0 left-0 h-screen w-screen bg-black/20"
      >
        <section
          class="relative mx-[10%] mt-[10%] flex flex-col bg-slate-50 xl:ml-[30%] "
        >
          <h3
            class="border-b-2 px-4 py-2 font-Taxicab text-xl"
          >
            Please confirm:
          </h3>
          <i
            class="fa-solid fa-xmark absolute top-2 right-4 text-xl hover:text-red-600"
          />
          <p
            class="px-6 py-4 text-lg text-black"
          >
            Are you sure you want to delete 
            <strong>
              Test
            </strong>
             and all its related information?
          </p>
          <div
            class="mt-6 flex flex-col justify-end gap-4 border-t-2 py-4 px-4 sm:flex-row"
          >
            <button
              class="flex items-center justify-center gap-2 rounded-md bg-green-600 px-4 py-2 font-semibold text-white hover:bg-green-300 hover:text-green-800 dark:text-white"
              type="button"
            >
              Confirm
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </button>
            <button
              class="flex items-center justify-center gap-2 rounded-md bg-red-600 px-4 py-2 font-semibold text-white hover:bg-red-300 hover:text-red-800"
              type="button"
            >
              Cancel
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CancelIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                />
              </svg>
            </button>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hidden absolute top-0 left-0 h-screen w-screen bg-black/20"
    >
      <section
        class="relative mx-[10%] mt-[10%] flex flex-col bg-slate-50 xl:ml-[30%] "
      >
        <h3
          class="border-b-2 px-4 py-2 font-Taxicab text-xl"
        >
          Please confirm:
        </h3>
        <i
          class="fa-solid fa-xmark absolute top-2 right-4 text-xl hover:text-red-600"
        />
        <p
          class="px-6 py-4 text-lg text-black"
        >
          Are you sure you want to delete 
          <strong>
            Test
          </strong>
           and all its related information?
        </p>
        <div
          class="mt-6 flex flex-col justify-end gap-4 border-t-2 py-4 px-4 sm:flex-row"
        >
          <button
            class="flex items-center justify-center gap-2 rounded-md bg-green-600 px-4 py-2 font-semibold text-white hover:bg-green-300 hover:text-green-800 dark:text-white"
            type="button"
          >
            Confirm
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckCircleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </button>
          <button
            class="flex items-center justify-center gap-2 rounded-md bg-red-600 px-4 py-2 font-semibold text-white hover:bg-red-300 hover:text-red-800"
            type="button"
          >
            Cancel
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CancelIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
              />
            </svg>
          </button>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Details View > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hidden absolute top-0 left-0 h-screen w-screen bg-black/20"
      >
        <section
          class="relative mx-[10%] mt-[10%] flex flex-col bg-slate-50 xl:ml-[30%] "
        >
          <h3
            class="border-b-2 px-4 py-2 font-Taxicab text-xl"
          >
            Please confirm:
          </h3>
          <i
            class="fa-solid fa-xmark absolute top-2 right-4 text-xl hover:text-red-600"
          />
          <p
            class="px-6 py-4 text-lg text-black"
          >
            Are you sure you want to delete 
            <strong>
              Test
            </strong>
             and all its related information?
          </p>
          <div
            class="mt-6 flex flex-col justify-end gap-4 border-t-2 py-4 px-4 sm:flex-row"
          >
            <button
              class="flex items-center justify-center gap-2 rounded-md bg-green-600 px-4 py-2 font-semibold text-white hover:bg-green-300 hover:text-green-800 dark:text-white"
              type="button"
            >
              Confirm
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </button>
            <button
              class="flex items-center justify-center gap-2 rounded-md bg-red-600 px-4 py-2 font-semibold text-white hover:bg-red-300 hover:text-red-800"
              type="button"
            >
              Cancel
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CancelIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                />
              </svg>
            </button>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hidden absolute top-0 left-0 h-screen w-screen bg-black/20"
    >
      <section
        class="relative mx-[10%] mt-[10%] flex flex-col bg-slate-50 xl:ml-[30%] "
      >
        <h3
          class="border-b-2 px-4 py-2 font-Taxicab text-xl"
        >
          Please confirm:
        </h3>
        <i
          class="fa-solid fa-xmark absolute top-2 right-4 text-xl hover:text-red-600"
        />
        <p
          class="px-6 py-4 text-lg text-black"
        >
          Are you sure you want to delete 
          <strong>
            Test
          </strong>
           and all its related information?
        </p>
        <div
          class="mt-6 flex flex-col justify-end gap-4 border-t-2 py-4 px-4 sm:flex-row"
        >
          <button
            class="flex items-center justify-center gap-2 rounded-md bg-green-600 px-4 py-2 font-semibold text-white hover:bg-green-300 hover:text-green-800 dark:text-white"
            type="button"
          >
            Confirm
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckCircleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </button>
          <button
            class="flex items-center justify-center gap-2 rounded-md bg-red-600 px-4 py-2 font-semibold text-white hover:bg-red-300 hover:text-red-800"
            type="button"
          >
            Cancel
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CancelIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
              />
            </svg>
          </button>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
